hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/ComputeStageImplBase.java

```diff
@@ -114,8 +115,9 @@
     @SuppressWarnings("unchecked")
     <RET> RET attachFilter(@Nonnull PredicateEx<T> filterFn) {
         checkSerializable(filterFn, "filterFn");
-        PredicateEx<T> adaptedFn = (PredicateEx<T>) fnAdapter.adaptFilterFn(filterFn);
-        return (RET) attach(new MapTransform<T, T>("filter", transform, t -> adaptedFn.test(t) ? t : null), fnAdapter);
+        PredicateEx<T> adaptedPred = (PredicateEx<T>) fnAdapter.adaptFilterFn(filterFn);
```

Comment:
wouldn't it be better to do the wrapping as part of the adaptFilterFn?

Reply:
I have moved the metrics related wrapping inside the function adapter. Looks indeed much better now, but had to make a bit of a compromise to do it for filter (due to adapting a predicate but needing a function in the end). I think this "compromise" thing I'm mentioning is the same thing your "have to do it before" comment is about too. Take a look at `ComputeStageImpBase.InternalFunctionAdapter.filterPredicateToFn`.